auth-api/src/auth_api/services/products.py

```diff
@@ -132,18 +137,33 @@ def create_product_subscription(org_id, subscription_data: Dict[str, Any],  # py
         return Product.get_all_product_subscription(org_id=org_id, skip_auth=True)
 
     @staticmethod
-    def _create_review_task(org, product_model, product_subscription, user):
+    def _subscribe_and_publish_activity(org_id, product_code, status_code, product_model_description):
+        subscription = ProductSubscriptionModel(org_id=org_id, product_code=product_code, status_code=status_code)\
+            .flush()
+        if status_code == ProductSubscriptionStatus.ACTIVE.value:
+            ActivityLogPublisher.publish_activity(Activity(org_id,
+                                                           ActivityAction.ADD_PRODUCT_AND_SERVICE.value,
+                                                           name=product_model_description))
+        return subscription
+
+    @staticmethod
+    def _create_review_task(org, product_model, product_subscription, user, external_source_id=None):
```

Comment:
alot of parameters here, might be worthwhile looking at dataclasses? 

Otherwise if someone wants to add a parameter, they're going into the danger zone with 5 parameters already on deck

Reply:
Updated to use data class, originally I kept the same signature with models but ran into a circular import issue. I ended up breaking it out to the specific fields